File: OCH_Tao/ABC/B/228.py
N,X = map(int,input().split())
L = [0]*N
L[X-1]=1
A = list(map(int,input().split()))
while True:
  X=A[X-1]
  if L[X-1]!=1:
    L[X-1]=1
  else:
    break
print(sum(L))

File: OCH_Tao/ABC/B/228.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [N, X] = input[0].split(" ").map(x => parseInt(x, 10));
let L = new Array(N).fill(0);
L[X - 1] = 1;
const A = input[1].split(" ").map(x => parseInt(x, 10));
let cnt = X;
while (true) {
  cnt = A[cnt - 1];
  if (L[cnt - 1] != 1) {
    L[cnt - 1] = 1;
  } else {
    break;
  }
}
console.log(L.reduce((a, b) => a + b, 0));